### PR TITLE
Update README with more information about Theia tests

### PR DESCRIPTION
**Pull request by katelynienaber**
Signed-off-by: chladnefazole <nienaber@protonmail.com>

## Proposed changes

I was trying to test Deepali's PR, and I couldn't run the tests without a lot of trouble. So, I added a few things to the docs:
- A list of dependencies you need for the tests
- A note to compile the code before running tests
- A note to make sure Theia is running in the background during test execution

## Release Notes

Milestone: Backlog

Changelog: Updated README with more information about Theia regression tests

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

File: docs/Developer's ReadMe.md [hunk 1]
```diff
@@ -123,25 +123,31 @@ Set up your Theia workspace for development purposes.
 
 **Follow these steps**:
 
-1. Build a VSIX file with your changes.
+1. Make sure you have the following installed:
 
-2. Build and run the Theia browser example, using [Setting up your Theia workspace](https://github.com/zowe/vscode-extension-for-zowe/blob/master/docs/README-Theia.md#setting-up-your-theia-workspace).
+   - [Mozilla Firefox](https://www.mozilla.org/en-US/firefox/new/)
+   - [Mozilla Geckodriver](https://github.com/mozilla/geckodriver/releases) (and add it to your PATH)
+   - [ChromeDriver](https://chromedriver.chromium.org/) (and add it to your PATH)
```
Comment (crawr) [inline docs/Developer's ReadMe.md:130]: I saw the current site will be deprecated soon. Perhaps we should point to the new site: https://sites.google.com/chromium.org/driver/ already?

**Review by crawr — COMMENTED** [review]
Thanks for adding these notes @katelynienaber ! I think this will definitely clear up how to run the Theia tests locally.

File: docs/Developer's ReadMe.md [hunk 2]
```diff
@@ -123,25 +123,31 @@ Set up your Theia workspace for development purposes.
 
 **Follow these steps**:
 
-1. Build a VSIX file with your changes.
+1. Make sure you have the following installed:
```
Comment (IgorCATech) [inline docs/Developer's ReadMe.md:126]: Ensure that you install the following components:

File: docs/Developer's ReadMe.md [hunk 3]
```diff
@@ -123,25 +123,31 @@ Set up your Theia workspace for development purposes.
...
 
-2. Build and run the Theia browser example, using [Setting up your Theia workspace](https://github.com/zowe/vscode-extension-for-zowe/blob/master/docs/README-Theia.md#setting-up-your-theia-workspace).
+   - [Mozilla Firefox](https://www.mozilla.org/en-US/firefox/new/)
+   - [Mozilla Geckodriver](https://github.com/mozilla/geckodriver/releases) (and add it to your PATH)
+   - [ChromeDriver](https://chromedriver.chromium.org/) (and add it to your PATH)
+   - Selenium webdriver with `npm i -g selenium-webdriver`
 
-   **Note**: Ensure that your latest VSIX file is in the `plugins` folder.
+2. Build a VSIX file with your changes.
```
Comment (IgorCATech) [inline docs/Developer's ReadMe.md:133]: Build a VSIX file that cointains your changes

File: docs/Developer's ReadMe.md [hunk 4]
```diff
@@ -123,25 +123,31 @@ Set up your Theia workspace for development purposes.
...
 
-3. Open a web browser and navigate to <http://localhost:3000> to verify your setup.
+3. Build and run the Theia browser example, using [Setting up your Theia workspace](https://github.com/zowe/vscode-extension-for-zowe/blob/master/docs/README-Theia.md#setting-up-your-theia-workspace).
+
+   - Ensure that your latest VSIX file is in the `plugins` folder.
+   - Theia must be **running** while you execute the tests!
```
Comment (IgorCATech) [inline docs/Developer's ReadMe.md:138]: Ensure that Theia runs (or 'is running' but we generally avoid any tense except Present Simple) while you execute the tests. 

Also, in tech.docs, we usually address the user directly, using the imperative mood

File: docs/Developer's ReadMe.md [hunk 5]
```diff
@@ -123,25 +123,31 @@ Set up your Theia workspace for development purposes.
...
+
+4. Open a web browser and navigate to <http://localhost:3000> to verify your setup.
```
Comment (IgorCATech) [inline docs/Developer's ReadMe.md:140]: If the point of this step is to verity the setup, I'd word it as follows: 
"(Optional) To verify your setup, open a browser and navigate to <http://localhost:3000>".

These verification steps are usually optional hence the `(optional)` marker

File: docs/Developer's ReadMe.md [hunk 6]
```diff
@@ -123,25 +123,31 @@ Set up your Theia workspace for development purposes.
...
 
    You should see Zowe Explorer that is deployed in Theia.
 
 ### Run Regression Tests
 
-Issue the following command to run the regression tests:
+1. Compile your project in VSCode with `yarn run compile`
```
Comment (IgorCATech) [inline docs/Developer's ReadMe.md:146]: Compile your project by running `yarn run compile` in your command-line interface

File: docs/Developer's ReadMe.md [hunk 7]
```diff
@@ -123,25 +123,31 @@ Set up your Theia workspace for development purposes.
...
 
-```
-yarn workspace vscode-extension-for-zowe test:theia
-```
+2. Run the Theia test with `yarn run test:theia`
```
Comment (IgorCATech) [inline docs/Developer's ReadMe.md:148]: Run the Theia test by issuing the following command:

```
yarn run test:theia
```

File: docs/Developer's ReadMe.md [hunk 8]
```diff
@@ -123,25 +123,31 @@ Set up your Theia workspace for development purposes.
...
 
-The regression tests output appears in your VSC debug console.
+The output from the tests will appear in your VSCode debug console.
```
Comment (IgorCATech) [inline docs/Developer's ReadMe.md:150]: The output from the tests appears in your VSCode debug console.

**Review by IgorCATech — COMMENTED** [review]
I haven't checked the factual part, only the stylistic one, and grammar. Left some minor comments. 
I would love to review the whole document at some point to see if more things can be improved

File: docs/Developer's ReadMe.md (hunk 1 above)
Comment (katelynienaber) [reply docs/Developer's ReadMe.md:130]: Yes, sounds like a good idea 👍 Also I'm adding Google Chrome here because both browsers are necessary, I think.

**Review by crawr — APPROVED** [review]
The content LGTM. Thanks @katelynienaber

**Review by jellypuno — CHANGES_REQUESTED** [review]

**Comment by jellypuno** [comment]
Another change that I would suggest is to make this document more formal. Re-arrange the sentence so you could avoid the usage of `you` or `your`

**Comment by IgorCATech** [comment]
We address the user directly, using the second-person pronouns, in all our docs. Using “you” pulls users into the information and makes it relevant to them. When you use “you” to address users, they are more likely to understand what their responsibility is.
This is one of the tech writing standards we adhere to.

**Comment by jellypuno** [comment]
@IgorCATech alright. If that is the rule then I'm fine with that.

**Review by jellypuno — APPROVED** [review]
Looks Good!

**Comment by deepali-hub** [comment]
Thanks @katelynienaber for updating this information and now it is more clear how to start with theia tests locally
LGTM!

**Review by deepali-hub — APPROVED** [review]

**Comment by jellypuno** [comment]
I added this in v1.15.0 release so we can add this change in CHANGELOG.md 
Thank you @katelynienaber for this

**Pull request merged**